tests/jballerina-unit-test/src/test/resources/test-src/object/client_resource_return_negative.bal

```diff
@@ -0,0 +1,119 @@
+            return successClient;
+        }
+
+        resource function get nullableClientReturnPath/[int a]/foo(string name) returns client object {}? {
+            return successClient;
+        }
+
+        resource function get nullableClientUnionFunctionReturnPath/[int a]/foo(string name)
+            returns (int|client object {} |
+                (function (int a, string b) returns int|string)|xml)? {
+            return testFunction;
+        }
+
+        resource function get nullableReadonlyClientReturnPath/[int a]/foo(string name)
+            returns client object {}? & readonly {
+            return successClient;
+        }
+    };
+}
+
+public function testErrorInResourceMethodReturningFunctionType() {
+    var successClientObject = client object {
+        resource function get .() returns function (int a, string b) returns int|string {
+            return testFunction;
+        }
+
+        resource function get readonlyFunctionReturnPath/[int a]/foo(string name) returns
+                function (int a, string b) returns int|string & readonly {
+            return testFunction;
+        }
+
+        resource function get nullableFunctionReturnPath/[int a]/foo(string name) returns
+                (function (int a, string b) returns int|string)? {
+            return testFunction;
+        }
+
+        resource function get nullableReadonlyFunctionReturnPath/[int a]/foo(string name) returns
+                (function (int a, string b) returns int|string)? & readonly {
+            return testFunction;
+        }
+    };
+}
+
+public function testErrorInResourceMethodReturningCustomObjectType() {
+    var successClientObject = client object {
+        resource function get .() returns CustomType1 {
+            return successClient;
+        }
+
+        resource function get clientUnionFunctionReturnPath/[int a]/foo(string name)
+                returns int|CustomType1|CustomType2 {
+            return testFunction;
+        }
+
+        resource function get readonlyClientReturnPath/[int a]/foo(string name) returns CustomType1 & readonly {
+            return successClient;
+        }
+    };
+}
+
+public function testErrorInResourceMethodReturningCustomFunctionType() {
+    var successClientObject = client object {
+        resource function get .() returns CustomType2 {
+            return testFunction;
+        }
+
+        resource function get readonlyFunctionReturnPath/[int a]/foo(string name) returns CustomType2 & readonly {
+            return testFunction;
+        }
+
+        resource function get nullableFunctionReturnPath/[int a]/foo(string name) returns CustomType2? {
+            return testFunction;
+        }
+
+        resource function get nullableReadonlyFunctionReturnPath/[int a]/foo(string name)
+                returns CustomType2? & readonly {
+            return testFunction;
+        }
+    };
+}
```

Comment:
Let's add a test case that we have both function type and client object type as a return type

Reply:
Added ebab2a74df0d4eddce0a69d7331556c045290bb9